driver/src/driver/stablex_driver.rs

```diff
@@ -41,8 +41,22 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
+        let submitted = if solution.is_non_trivial() {
+            // NOTE: in retrieving the objective value from the contract the
+            //   solution gets validated, ensured that it is better than the
+            //   latest submitted solution, and that solutions are still being
+            //   accepted for this batch ID.
+            let objective_value = self.contract.get_solution_objective_value(
+                batch,
+                orders.clone(),
+                solution.clone(),
+            )?;
```

Comment:
The print statement below is no longer accurate. You might not submit a non_trivial solution ("Not submitting trivial solution for batch {}")

Reply:
The `println!("Not submitting trivial solution...")` should only gets called if solution is trivial.

Reply:
The `println!("Not submitting trivial solution...")` still only gets called if solution is trivial.